relay/blockchain/currency_network_proxy.py

```diff
@@ -147,33 +147,45 @@ def log(log_entry):
         self.start_listen_on(TransferEventType, log)
 
     def get_network_events(self, event_name: str, user_address: str=None, from_block: int=0) -> List[BlockchainEvent]:
+        logger.debug("get_network_events: event_name=%s user_address=%s from_block=%s",
+                     event_name,
+                     user_address,
+                     from_block)
         if user_address is None:
-            result = self.get_events(event_name, from_block=from_block)
+            events = self.get_events(event_name, from_block=from_block)
         else:
 
             filter1 = {from_to_types[event_name][0]: user_address}
             filter2 = {from_to_types[event_name][1]: user_address}
 
-            events = [
+            spawned_greenlets = [
                 gevent.spawn(self.get_events, event_name, filter1, from_block),
                 gevent.spawn(self.get_events, event_name, filter2, from_block)
             ]
-            gevent.joinall(events, timeout=2)
-            result = list(itertools.chain.from_iterable([event.value for event in events]))
-
-            for event in result:
+            finished_greenlets = gevent.joinall(spawned_greenlets, raise_error=True, timeout=15)
+            if len(finished_greenlets) < len(spawned_greenlets):
```

Comment:
This should not only produce a warning, but should fail. We were clearly not able to perform the requested operation.